lib/bier/query_parser.ex

```diff
@@ -0,0 +1,354 @@
+  # underscore, so identifiers of this form are safe against possible conflict
+  # with future extensions of the standard.
+  #
+  # See: https://www.postgresql.org/docs/9.2/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
+  # FIXME: Complete this based on previous description
+  identifier =
+    [?_, ?A..?Z, ?a..?z]
+    |> ascii_char()
+    |> repeat(ascii_char([?_, ?a..?z, ?A..?Z, ?0..?9]))
+
+  whitespace =
+    ascii_char([
+      @horizontal_tab,
+      @space
+    ])
+
+  default_select =
+    ascii_char([?*])
+    |> concat(eos())
+    |> tag(:default)
+
+  column_separator = ascii_char([?,])
+  alias_separator = ascii_char([?:])
+  cast_separator = times(ascii_char([?:]), 2)
+
+  column_alias =
+    identifier
+    |> lookahead_not(cast_separator)
+    |> concat(ignore(alias_separator))
+    |> tag(:alias)
+
+  # TODO: Complete the following list
+  casting_types =
+    choice([
+      string("boolean"),
+      string("date"),
+      string("float"),
+      string("integer"),
+      string("interval"),
+      string("text"),
+      string("timestamp")
+    ])
+
+  column_cast =
+    cast_separator
+    |> ignore()
+    |> concat(casting_types)
+    |> unwrap_and_tag(:cast)
+
+  column =
+    column_alias
+    |> optional()
+    |> concat(tag(identifier, :column))
+    |> optional(column_cast)
+
+  detailed_select =
+    column
+    |> optional(ignore(optional(column_separator, whitespace)))
+    |> post_traverse(:join_and_wrap)
+    |> times(min: 1)
+    |> eos()
+
+  defp join_and_wrap(_rest, args, context, _line, _offset) do
+    # TODO: Probably is a good idea to pass the column, cast, and column alias
+    # as a data structure instead of a string, that way, a following step could
+    # verify if the given column actually exist in the given table, but first, we need
+    # to introspect this information from the DB.
+    column = Keyword.fetch!(args, :column)
+    cast = Keyword.get(args, :cast)
+    column_alias = Keyword.get(args, :alias)
+
+    result =
+      cond do
+        cast && column_alias ->
+          ~s|#{column}::#{cast} AS "#{column_alias}"|
+
+        cast ->
+          "#{column}::#{cast}"
+
+        column_alias ->
+          ~s|#{column} AS "#{column_alias}"|
+
+        true ->
+          to_string(column)
+      end
+
+    {List.wrap(result), context}
+  end
+
+  defparsecp(:select, choice([default_select, detailed_select]))
+
+  @doc """
+  Parse the given `select` query string
+
+  ## Examples
+
+      iex> parse_select("*")
+      {:ok, "*"}
+      iex> parse_select("first_name,age")
+      {:ok, "first_name, age"}
+      iex> parse_select("fullName:full_name,birthDate:birth_date")
+      {:ok, ~S/full_name AS "fullName", birth_date AS "birthDate"/}
+      iex> parse_select("uno:first::text, dos:second, third, forth::text")
+      {:ok, ~S/first::text AS "uno", second AS "dos", third, forth::text/}
+  """
+  @spec parse_select(String.t()) :: {:ok, String.t()} | {:error, String.t()}
```

Comment:
* [ ] TODO: Re-evaluate this; instead of returning a string, return a list of objects to be consistent with the other function, and leave the concern of adapting the query down the road, possibly in the executor.